mex.go

```diff
@@ -228,22 +237,29 @@ func (mex *messageExchange) recvPeerFrameOfType(msgType messageType) (*Frame, er
 	}
 }
 
+func (mex *messageExchange) onCtxErr(err error) {
+	if onCancel := mex.mexset.onCancel; onCancel != nil {
+		onCancel(mex.msgID)
+	}
+}
+
 // shutdown shuts down the message exchange, removing it from the message
 // exchange set so  that it cannot receive more messages from the peer.  The
 // receive channel remains open, however, in case there are concurrent
 // goroutines sending to it.
-func (mex *messageExchange) shutdown() {
+func (mex *messageExchange) shutdown() bool {
```

Comment:
the return value doesn't seem to be used anywhere

Reply:
Good catch, this was from the old PR which I started off, no longer need this so reverted.